tests/__init__.py

```diff
@@ -1,40 +1,65 @@
-from typing import Any, Dict, Optional
+from typing import Any, Dict
 
-import attr
 from mock import Mock
 from synapse.module_api import ModuleApi
 
 from threepid_checker import ThreepidChecker
 
 
-@attr.s(auto_attribs=True)
-class MockEvent:
-    """Mocks an event. Only exposes properties the module uses."""
-    sender: str
-    type: str
-    content: Dict[str, Any]
-    room_id: str = "!someroom"
-    state_key: Optional[str] = None
+class Addresses:
+    """Known addresses for tests."""
 
-    def is_state(self) -> bool:
-        """Checks if the event is a state event by checking if it has a state key."""
-        return self.state_key is not None
+    NO_HS = "nohs@test"
+    WRONG_HS = "wronghs@test"
+    REQUIRES_INVITE_MISSING = "rimissing@test"
+    NO_INVITE_NEEDED = "dontinvite@test"
+    INVITED_MISSING = "invitedmissing@test"
+    NOT_INVITED = "notinvited@test"
+    INVITED = "invited@test"
+    INVITED_NOT_REQUIRED = "invitednotrequired@test"
 
-    @property
-    def membership(self) -> str:
-        """Extracts the membership from the event. Should only be called on an event
-        that's a membership event, and will raise a KeyError otherwise.
+
+class MockHttpClient:
+    async def get_json(self, url: str, params: Dict[str, Any]) -> Dict[str, Any]:
+        """Mocks the get_json method of Synapse's HTTP client.
+
+        Args:
+            url: The URL to query. We're ignoring it for tests.
+            params: The parameters for the request.
+
+        Returns:
+            A dict which content depends on the 3PID address passed in the parameters.
+
+        Raises:
+            RuntimeError if the provided address isn't known.
         """
-        membership: str = self.content["membership"]
-        return membership
+        address = params["address"]
+        if address == Addresses.NO_HS:
```

Comment:
This feels like it would work better as a mapping of some sort, but note a huge deal for test code. 👍

Reply:
Yeah I thought about that, but thought doing it with ifs and elses was a bit more readable, albeit more boilerplatey.